src/index.ts

```diff
@@ -533,6 +533,19 @@ export async function createServer() {
     type: 'onRequest',
     method(request: Hapi.Request & { sentryScope?: any }, h) {
       request.sentryScope?.setExtra('payload', request.payload)
+      const coreVersion = request.headers['x-version']
+      const countryConfigVersion = process.env.npm_package_version
+      if (
+        coreVersion &&
+        coreVersion !== 'undefined' &&
+        countryConfigVersion &&
+        countryConfigVersion !== 'undefined' &&
+        coreVersion !== countryConfigVersion
+      ) {
+        const errorMessage = `Version mismatch!! Opencrvs-Core is running on: ${coreVersion}, Opencrvs-Countryconfig is running on: ${countryConfigVersion}`
+        console.error(errorMessage)
+        return h.response({ message: errorMessage }).code(426).takeover()
+      }
```

Comment:
```suggestion
```

I'm thinking we could make an assumption here that the client is always up to date as its core's responsibility to verify it. One of our design goals is to hide country config completely behind gateway which would also eliminate this issue